scripts/add_extra_components.py

```diff
@@ -179,15 +221,75 @@ def attach_stores(n, costs, config):
             bus0=b_buses_i,
             bus1=buses_i,
             carrier="battery discharger",
+            # NB: the efficiencies are "round trip efficiencies"
             efficiency=costs.at["battery inverter", "efficiency"],
             p_nom_extendable=True,
             marginal_cost=costs.at["battery inverter", "marginal_cost"],
         )
 
+    for c in carriers_database:
+        carrier_buses_i = n.madd("Bus", buses_i + f" {c}", carrier=c, **bus_sub_dict)
+        tech_type = costs.technology_type
+        carrier = costs.carrier
+        store_filter = (carrier == c) & (tech_type == "store")
+        charger_or_bicharger_filter = (carrier == c) & (
+            (tech_type == "charger") | (tech_type == "bicharger")
+        )
+        discharger_or_bicharger_filter = (carrier == c) & (
+            (tech_type == "discharger") | (tech_type == "bicharger")
```

Comment:
these may be reformulated into something like `tech_type in ["discharger", "bicharger"]`

Reply:
not working immediately, I recommend coherent filtering as it is